fuel/app/classes/materia/api/v1.php

```diff
@@ -125,6 +167,7 @@ static public function widget_instance_new($widget_id=null, $name=null, $qset=nu
 			'guest_access'    => $is_student,
 			'attempts'        => -1
 		]);
+		if ( ! $is_draft && ! $widget->publishable_by(\Model_User::find_current_id()) ) return new Msg(Msg::ERROR, 'Widget type can not be published by students.');
```

Comment:
I'd do this as soon as it's possible - move up to after `widget->get()` is called